psyneulink/library/components/mechanisms/processing/transfer/lstmmechanism.py

```diff
@@ -0,0 +1,424 @@
+# Princeton University licenses this file to You under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.  You may obtain a copy of the License at:
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+# ****************************************  LSTMMechanism *************************************************
+
+import numpy as np
+import typecheck as tc
+
+from collections.abc import Iterable
+
+from psyneulink.core import llvm as pnlvm
+from psyneulink.core.components.functions.transferfunctions import Logistic, Tanh
+from psyneulink.core.components.functions.userdefinedfunction import UserDefinedFunction
+from psyneulink.core.components.mechanisms.processing.processingmechanism import ProcessingMechanism
+from psyneulink.core.components.projections.pathway.mappingprojection import MappingProjection
+from psyneulink.core.components.ports.inputport import InputPort
+from psyneulink.core.globals.keywords import IDENTITY_MATRIX, LSTM_MECHANISM, OWNER_VALUE, RESULT
+from psyneulink.core.globals.parameters import Parameter
+from psyneulink.core.components.ports.outputport import OutputPort
+
+__all__ = [
+    'LSTMMechanism'
+]
+
+RECURRENT_INPUT_PORT_NAME = "RECURRENT_INPUT"
+RECURRENT_OUTPUT_PORT_NAME = 'RECURRENT_OUTPUT'
+CELL_STATE_INPUT_PORT_NAME = 'CELL_STATE_INPUT'
+CELL_STATE_OUTPUT_PORT_NAME = 'CELL_STATE_OUTPUT'
+EXTERNAL = 'EXTERNAL'
+
+class LSTMMechanism(ProcessingMechanism):
```

Comment:
The implementations differ, since the LSTM mechanism implementation aims to remove dependence on AutoAssociativeProjections.

It may make more sense to rework RNNs first, and then inherit from RNNs.

Reply:
ah, ok. It was more of my curiosity, I don't have a strong opinion about using RTM.
handling the functions and ports in a standard way is imo more important.